pkg/virt-controller/watch/vm.go

```diff
@@ -336,15 +340,34 @@ func (c *VMController) execute(key string) error {
 	return syncErr
 }
 
-func (c *VMController) authorizeDataVolume(vm *virtv1.VirtualMachine, dataVolume *cdiv1.DataVolume) error {
-	if dataVolume.Spec.SourceRef != nil {
+func (c *VMController) handleCloneDataVolume(vm *virtv1.VirtualMachine, dv *cdiv1.DataVolume) error {
+	if dv.Spec.SourceRef != nil {
 		return fmt.Errorf("DataVolume sourceRef not supported")
 	}
 
-	if dataVolume.Spec.Source == nil || dataVolume.Spec.Source.PVC == nil {
+	if dv.Spec.Source == nil || dv.Spec.Source.PVC == nil {
 		return nil
 	}
 
+	// For consistency with other k8s objects, we allow creating clone DataVolumes even when the source PVC doesn't exist.
+	// This means that a VirtualMachine can be successfully created with volumes that may remain unpopulated until the source PVC is created.
+	// For this reason, we check if the source PVC exists and, if not, we trigger an event to let users know of this behavior.
+	_, err := c.clientset.CoreV1().PersistentVolumeClaims(dv.Spec.Source.PVC.Namespace).Get(context.Background(), dv.Spec.Source.PVC.Name, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	} else if errors.IsNotFound(err) {
+		c.recorder.Eventf(vm, k8score.EventTypeWarning, SourcePVCNotAvailabe, "Source PVC %s not available: DataVolume %s will remain unpopulated until source PVC is created", dv.Spec.Source.PVC.Name, dv.Name)
```

Comment:
I think `DataVolume %s will remain unpopulated` should be rephrased to something like `Target PVC %s will remain unpopulated`.